templates/init/functions/typescript/tslint.json

```diff
@@ -1,115 +0,0 @@
-{
-  "rules": {
-    // -- Strict errors --
-    // These lint rules are likely always a good idea.
-
-    // Force function overloads to be declared together. This ensures readers understand APIs.
-    "adjacent-overload-signatures": true,
-
-    // Do not allow the subtle/obscure comma operator.
-    "ban-comma-operator": true,
-
-    // Do not allow internal modules or namespaces . These are deprecated in favor of ES6 modules.
-    "no-namespace": true,
-
-    // Do not allow parameters to be reassigned. To avoid bugs, developers should instead assign new values to new vars.
-    "no-parameter-reassignment": true,
-
-    // Force the use of ES6-style imports instead of /// <reference path=> imports.
-    "no-reference": true,
-
-    // Do not allow type assertions that do nothing. This is a big warning that the developer may not understand the
-    // code currently being edited (they may be incorrectly handling a different type case that does not exist).
-    "no-unnecessary-type-assertion": true,
-
-    // Disallow nonsensical label usage.
-    "label-position": true,
-
-    // Disallows the (often typo) syntax if (var1 = var2). Replace with if (var2) { var1 = var2 }.
-    "no-conditional-assignment": true,
-
-    // Disallows constructors for primitive types (e.g. new Number('123'), though Number('123') is still allowed).
-    "no-construct": true,
-
-    // Do not allow super() to be called twice in a constructor.
-    "no-duplicate-super": true,
-
-    // Do not allow the same case to appear more than once in a switch block.
-    "no-duplicate-switch-case": true,
-
-    // Do not allow a variable to be declared more than once in the same block. Consider function parameters in this
-    // rule.
-    "no-duplicate-variable": [true, "check-parameters"],
```

Comment:
.eslintrc.js doesn't have this "check-parameters" flag on "no-redeclare" - will it still check params? Is there a way to add this to .eslintrc.js?

Reply:
I examined if 'no-redeclare' rule can check for variables with the same name as a parameter.

The test code is
```js
function foo(bar: string): void {
  console.log(bar);
  const bar = 1;
  return;
}
```

The result is
```
> eslint src "--fix"


~/firebase-tools/functions/src/index.ts
  13:9  error  'bar' is already defined  no-redeclare

✖ 1 problem (1 error, 0 warnings)
```

Reply:
Great, thanks for confirming!